internal/uvm/start.go

```diff
@@ -141,12 +151,35 @@ func (uvm *UtilityVM) Start(ctx context.Context) (err error) {
 	ctx, cancel := context.WithTimeout(ctx, 2*time.Minute)
 	defer cancel()
 
+	g, gctx := errgroup.WithContext(ctx)
+
+	// Prepare to provide entropy to the init process in the background. This
+	// must be done in a goroutine since, when using the internal bridge, the
+	// call to Start() will block until the GCS launches, and this cannot occur
+	// until the host accepts and closes the entropy connection.
+	if uvm.entropyListener != nil {
+		g.Go(func() error {
+			conn, err := uvm.acceptAndClose(gctx, uvm.entropyListener)
+			uvm.entropyListener = nil
```

Comment:
We should also handle the case where we close the uvm before this finishes? https://github.com/microsoft/hcsshim/blob/master/internal/uvm/create.go#L139

Reply:
What I mean is if we hit a code path where we issue a `Create(); Close()` without ever calling start.

Reply:
I reshuffled some things and cleaned up the log handling stuff too to follow the same pattern.